dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/managegroups/ManageGroupsCRUDServiceIT.java

```diff
@@ -0,0 +1,136 @@
+
+    @PersistenceContext(unitName = "VDCNet-ejbPU")
+    private EntityManager em;
+
+    @EJB
+    private ManageGroupsCRUDService manageGroupsCRUDService;
+    @EJB
+    private DataverseServiceBean dataverseService;
+    @Inject
+    private DataverseSession dataverseSession;
+    @EJB
+    private AuthenticationServiceBean authenticationService;
+    @EJB
+    private ExplicitGroupServiceBean explicitGroupService;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationService.getAdminUser());
+    }
+
+    @Test
+    public void shouldCreateGroup() {
+        // given
+        Dataverse dv = dataverseService.findByAlias("ownmetadatablocks");
+        RoleAssignee roleAssignee = authenticationService.findAllAuthenticatedUsers().get(0);
+
+        // when
+        manageGroupsCRUDService.create(dv, "testGroup", "testGroupId", "testDesc", Collections.singletonList(roleAssignee));
+
+        // then
+        ExplicitGroup dbExplicitGroup = explicitGroupService.findByAlias(dv.getId() + "-testGroupId");
+
+        Assert.assertEquals(1, explicitGroupService.findByOwner(dv.getId()).size());
+        Assert.assertEquals("testGroup", dbExplicitGroup.getDisplayName());
+        Assert.assertEquals(roleAssignee, dbExplicitGroup.getContainedAuthenticatedUsers().iterator().next());
+    }
+
+    @Test
+    public void shouldUpdateGroup() {
+        // given
+        Dataverse dv = dataverseService.findByAlias("ownmetadatablocks");
+        ExplicitGroup explicitGroup = buildExplicitGroup(dv);
+
+        em.persist(explicitGroup);
+
+        // when
+        explicitGroup.setDisplayName("updatedName");
+
+        AuthenticatedUser newRoleAssignee = authenticationService.findAllAuthenticatedUsers().get(1);
+        Set<AuthenticatedUser> newRoleAssigneesSet = explicitGroup.getContainedAuthenticatedUsers();
+        newRoleAssigneesSet.add(newRoleAssignee);
+        List<RoleAssignee> newRoleAssigneesList = new LinkedList<>(newRoleAssigneesSet);
+
+        manageGroupsCRUDService.update(explicitGroup, newRoleAssigneesList);
+
+        // then
+        ExplicitGroup dbExplicitGroup = explicitGroupService.findByAlias(dv.getId() + "-explicitGroupIdentifier");
+        Assert.assertEquals(1, explicitGroupService.findByOwner(dv.getId()).size());
+        Assert.assertEquals(explicitGroup.getId(), dbExplicitGroup.getId());
+        Assert.assertEquals("updatedName", dbExplicitGroup.getDisplayName());
+        Assert.assertNotEquals("explicitGroupName", dbExplicitGroup.getDisplayName());
+        Assert.assertEquals(2, dbExplicitGroup.getContainedAuthenticatedUsers().size());
+        Assert.assertEquals(newRoleAssignee.getIdentifier(),
+                dbExplicitGroup.getContainedAuthenticatedUsers()
+                        .stream()
+                        .filter(au -> au.getIdentifier().equals(newRoleAssignee.getIdentifier()))
+                        .findAny().get().getIdentifier()
+        );
```

Comment:
This condition is more complicated than it needs to be (and in case of error it won't reach assertEquals - it will throw error on `.get()`).
```
List<String> dbContainedAuthenticatedUserIdentifiers = dbExplicitGroup.getContainedAuthenticatedUsers().stream().map(u -> u.getIdentifier()).collect(toList());
assertThat(dbContainedAuthenticatedUserIdentifiers, hasItem(newRoleAssignee.getIdentifier()));
```